### PR TITLE
feat(#major); gains-trade; add gains-trade subgraph 

### PR DESCRIPTION
**Pull request by dhruv-chauhan**
**Description**

- Added gains-trade subgraph using perpfutures schema.
- Added configs for networks: matic, arbitrum

**Links**

- polygon:
  - https://thegraph.com/hosted-service/subgraph/dhruv-chauhan/gains-trade-polygon
  - https://okgraph.xyz/?q=dhruv-chauhan%2Fgains-trade-polygon
  - https://subgraphs.messari.io/subgraph?endpoint=dhruv-chauhan/gains-trade-polygon&tab=protocol
- arbitrum:
  - https://thegraph.com/hosted-service/subgraph/dhruv-chauhan/gains-trade-arbitrum
  - https://okgraph.xyz/?q=dhruv-chauhan%2Fgains-trade-arbitrum
  - https://subgraphs.messari.io/subgraph?endpoint=dhruv-chauhan/gains-trade-arbitrum&tab=protocol

**New Subgraph Checklist**

- [x] Researched protocol / contracts
- [x] Modeled off this [template](https://github.com/messari/subgraphs/tree/master/subgraphs/_reference_) using the latest versions
- [x] Updated title
- [x] Added subgraph(s) to [deployment.json](https://github.com/messari/subgraphs/blob/master/deployment/deployment.json)
- [x] Used standard subgraph libraries (if necessary/available)
- [x] Viewed subgraph in https://subgraphs.messari.io/ to ensure no ERRORS from the automatic data validation / warnings are reasonable
- [x] General aggregate metrics can be similar to known sources like the protocol or defi llama (The detailed QA will be done by a protocol specialist)
- [x] Assigned the proper reviewers. If you are unsure, ask in our discord!
- [x] Followed docs, especially [`WALKTHROUGH.md`](https://github.com/messari/subgraphs/blob/master/docs/WALKTHROUGH.md)

### CODE REVIEW

**Comment by dhruv-chauhan** [comment]
> #1924
> 
> Just double checking, there was an update to the perpetuals schema. Making sure you have it :)

updated to latest! @ishraq8

File: subgraphs/gains-trade/src/mappings/handlers.ts [hunk 1]
```diff
@@ -0,0 +1,849 @@
...
+    }
+
+    position.setLeverage(leverage.toBigDecimal());
+    position.setBalance(collateralToken, collateralAmount);
+    position.setCollateralBalance(collateralToken, collateralAmount);
+    position.addCollateralInCount();
+    position.setFundingrateOpen(fundingRatePerDay);
+  } else {
+    const percentProfit = event.params.percentProfit;
+    const pnl = safeDivide(
```
Comment (dmelotik) [inline subgraphs/gains-trade/src/mappings/handlers.ts:657]: same thing

**Review by dmelotik — COMMENTED** [review]
Nice work on this @dhruv-chauhan !

My comments on the subgraph code/config will live in this PR, my comments on the SDK changes/schema changes will be in the other PR

**Comment by dhruv-chauhan** [comment]
@dmelotik 
- added README
- MarketExecuted / LimitExecuted should be much easier to read now
- implemented suggestions

File: subgraphs/gains-trade/src/mappings/helpers.ts [hunk 2]
```diff
@@ -25,3 +48,279 @@ export function createTokenAmountArray(
...
+      pairIndex.toString(),
+    ]);
+    return;
+  }
+  const closeFeeP = closeFeePCall.value;
+  const closingFee = bigDecimalToBigInt(
+    leveragedAmount
+      .times(closeFeeP)
+      .toBigDecimal()
+      .div(PRECISION.times(BIGINT_HUNDRED).toBigDecimal())
```
Comment (dmelotik) [inline subgraphs/gains-trade/src/mappings/helpers.ts:248]: Maybe `PRECISION.times(BIGINT_HUNDRED).toBigDecimal()` should just be stored as `PRECISION_BD` in cosntants.ts?

**Review by dmelotik — COMMENTED** [review]
Code changes looking good? Like Ishraq said [here](https://github.com/messari/subgraphs/issues/2005#issuecomment-1520796673) should we just use the latest schema since the SDK supports that?

**Review by dmelotik — APPROVED** [review]
Approving, just waiting to merge once ishraq's final green light is given on the funding rate fix and the schema update

**Comment by ishraq8** [comment]
Green light for arbitrum and polygon. this can be merged and hosted.

**Pull request merged**